README.md

```diff
@@ -270,12 +271,12 @@ generated_qpc_path = QEfficient.compile(
 Benchmark the model on Cloud AI 100, run the infer API to print tokens and tok/sec
 
 ```Python
-from QEfficient.generation.text_generation_inference import get_compilation_batch_size
+from QEfficient.generation.text_generation_inference import get_compilation_dims
 
 # post compilation, we can print the latency stats for the kv models, We provide API to print token and Latency stats on AI 100
 # We need the compiled prefill and decode qpc to compute the token generated, This is based on Greedy Sampling Approach
-batch_size = get_compilation_batch_size(generated_qpc_path)
-QEfficient.cloud_ai_100_exec_kv(batch_size=batch_size, tokenizer=tokenizer, qpc_path=generated_qpc_path, device_id=[0], prompt=["My name is"])
+batch_size, ctx_len = get_compilation_dims(generated_qpc_path)
```

Comment:
can we move this call inside `cloud_ai_100_exec_kv`?
so, we only ask users for `tokenizer`, `qpc_path` and `device_id`

Reply:
If we move this inside cloud_ai_100_exec_kv then we need to move the check_num_prompts_and_batch_size also inside cloud_ai_100_exec_kv, (for execute API)

Then in case of infer API, it will cause check_num_prompts_and_batch_size being called twice, if we remove the first call of this function from infer API then it will export and compile but fail at last step of execution if there is mismatch in num_prompts and batch_size.

This is what we discussed last time.

Reply:
I don't understand, why do you say we need to move `check_num_prompts_and_batch_size` for this?

Reply:
because input of this function is compilation_batch_size which is output of get_compilation_dims

Reply:
As discussed, let's separate 
1. checking `batch_size` with `num_of_prompts` 
2. figuring out prompts from `prompt` and `prompt_txt_file`

So, that we can call `func` 1 twice, and it won't lead to increased runtime, and we can only expose users to the HL API which is exec as explained in first comment.

Reply:
Added the suggensted changes to PR#47.